src/borg/testsuite/__init__.py

```diff
@@ -235,26 +235,63 @@ def _assert_dirs_equal_cmp(self, diff, ignore_bsdflags=False, ignore_xattrs=Fals
             self._assert_dirs_equal_cmp(sub_diff, ignore_bsdflags=ignore_bsdflags, ignore_xattrs=ignore_xattrs, ignore_ns=ignore_ns)
 
     @contextmanager
-    def fuse_mount(self, location, mountpoint, *options):
-        os.mkdir(mountpoint)
-        args = ['mount', location, mountpoint] + list(options)
-        self.cmd(*args, fork=True)
+    def fuse_mount(self, location, mountpoint=None, *options, **kwargs):
+        if mountpoint is None:
+            mountpoint = tempfile.mkdtemp()
+        else:
+            os.mkdir(mountpoint)
+        if 'fork' not in kwargs:
+            # For a successful mount, `fork = True` is required. The
+            # option to run the command with `fork = False` is still
+            # necessary to test mount failures like attempting to mount
+            # a read-only repo.
+            kwargs['fork'] = True
```

Comment:
I guess it has something to do with running the daemon that `fork=True` is required?

Reply:
guess so.

Reply:
:+1: I have improved the explanation a little bit. This should now clear up any confusion, why the `fork` argument is handled so particulary.